src/metalperformanceshadersgraph.cs

```diff
@@ -0,0 +1,1831 @@
+	[BaseType (typeof(MPSGraph))]
+	interface MPSGraph_MPSGraphTopKOps
+	{
+		// -(NSArray<MPSGraphTensor *> * _Nonnull)topKWithSourceTensor:(MPSGraphTensor * _Nonnull)source k:(NSUInteger)k name:(NSString * _Nullable)name __attribute__((swift_name("topK(_:k:name:)")));
+		[Export ("topKWithSourceTensor:k:name:")]
+		MPSGraphTensor[] TopK (MPSGraphTensor source, nuint k, [NullAllowed] string name);
+
+		// -(NSArray<MPSGraphTensor *> * _Nonnull)topKWithSourceTensor:(MPSGraphTensor * _Nonnull)source kTensor:(MPSGraphTensor * _Nonnull)kTensor name:(NSString * _Nullable)name __attribute__((swift_name("topK(_:kTensor:name:)")));
+		[Export ("topKWithSourceTensor:kTensor:name:")]
+		MPSGraphTensor[] TopK (MPSGraphTensor source, MPSGraphTensor kTensor, [NullAllowed] string name);
+	}
+
+	// @interface MPSGraphTopKGradientOps (MPSGraph)
+	[Category]
+	[BaseType (typeof(MPSGraph))]
+	interface MPSGraph_MPSGraphTopKGradientOps
+	{
+		// -(MPSGraphTensor * _Nonnull)topKWithGradientTensor:(MPSGraphTensor * _Nonnull)gradient source:(MPSGraphTensor * _Nonnull)source k:(NSUInteger)k name:(NSString * _Nullable)name __attribute__((swift_name("topKGradient(_:input:k:name:)")));
+		[Export ("topKWithGradientTensor:source:k:name:")]
+		MPSGraphTensor TopKGradient (MPSGraphTensor gradient, MPSGraphTensor source, nuint k, [NullAllowed] string name);
+
+		// -(MPSGraphTensor * _Nonnull)topKWithGradientTensor:(MPSGraphTensor * _Nonnull)gradient source:(MPSGraphTensor * _Nonnull)source kTensor:(MPSGraphTensor * _Nonnull)kTensor name:(NSString * _Nullable)name __attribute__((swift_name("topKGradient(_:input:kTensor:name:)")));
+		[Export ("topKWithGradientTensor:source:kTensor:name:")]
+		MPSGraphTensor TopKGradient (MPSGraphTensor gradient, MPSGraphTensor source, MPSGraphTensor kTensor, [NullAllowed] string name);
+	}
+
+	// @interface MPSGraphCompilationDescriptor : NSObject
+	[TV (15,0), Mac (12,0), iOS (15,0), MacCatalyst (15,0)]
+	[BaseType (typeof(NSObject))]
+	interface MPSGraphCompilationDescriptor
+	{
+		// -(void)disableTypeInference;
+		[Export ("disableTypeInference")]
+		void DisableTypeInference ();
+	}
+
+	// @interface MPSGraphDevice : NSObject
+	[iOS (14,0), TV (14,0), Mac (11,0), MacCatalyst (14,0)]
+	[BaseType (typeof(NSObject))]
+	interface MPSGraphDevice
+	{
+		// @property (readonly, nonatomic) MPSGraphDeviceType type;
+		[Export ("type")]
+		MPSGraphDeviceType Type { get; }
+
+		// @property (readonly, nonatomic) id<MTLDevice> _Nullable metalDevice;
+		[NullAllowed, Export ("metalDevice")]
+		IMTLDevice MetalDevice { get; }
+
+		// +(instancetype _Nonnull)deviceWithMTLDevice:(id<MTLDevice> _Nonnull)metalDevice;
+		[Static]
+		[Export ("deviceWithMTLDevice:")]
+		MPSGraphDevice FromMTLDevice (IMTLDevice metalDevice);
+	}
+
+	// typedef void (^MPSGraphExecutableCompletionHandler)(NSArray<MPSGraphTensorData *> * _Nonnull, NSError * _Nullable);
+	delegate void MPSGraphExecutableCompletionHandler (MPSGraphTensorData[] arg0, [NullAllowed] NSError error);
```

Comment:
Can the parameter name be improved here?

```suggestion
	delegate void MPSGraphExecutableCompletionHandler (MPSGraphTensorData[] results, [NullAllowed] NSError error);
```